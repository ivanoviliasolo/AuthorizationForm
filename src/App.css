.App {
  font-size: 2em;
  color: red;
}

/*# sourceMappingURL=App.css.map */
