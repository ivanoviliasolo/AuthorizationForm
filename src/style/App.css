.main-dialog-background {
  position: absolute;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
}

.main-dialog-window {
  text-align: center;
  position: relative;
  top: 30vh;
  left: 30vw;
  height: 40vh;
  width: 40vw;
  background-color: burlywood;
}

.dialog-header {
  background-color: sandybrown;
  padding: 10px;
  height: 2vh;
}

.header-close-button {
  float: right;
}

.header-close-button:hover {
  cursor: pointer;
  background-color: cadetblue;
}

.title-of-input {
  padding: 5px;
  text-align: end;
}

.wrapper-div-of-input {
  padding: 5px 0.5vw;
  width: 39vw;
}

.field-with-data-of-input {
  width: 50%;
  text-align: center;
  float: right;
  margin-right: 20px;
}

.wrapper-near-the-all-fields {
  height: 35vh;
  overflow-y: auto;
  overflow-x: hidden;
  background: blueviolet;
}

.main-dialog-footer {
  text-align: end;
  padding: 5px;
  background: sandybrown;
  height: 3vh;
}

.main-dialog-footer-indent {
  margin-right: 10px;
}

.error-message {
  text-align: center;
  margin-top: 2vh;
}

.users-error-footer {
  float: left;
}

.wait-request {
  position: absolute;
  top: 15vh;
  left: 10vw;
  padding: 2vw;
  background-color: #2c6370;
  border-radius: 1vw;
}

.wrapp-main {
  position: absolute;
  top: 0px;
  left: 0px;
  width: 100vw;
  height: 100vh;
  padding: 20vw;
}

.unlogin-button {
  margin-left: 1vw;
}

div {
  color: black;
  background-color: rgba(172, 179, 184, 0.5);
}

/*# sourceMappingURL=App.css.map */
