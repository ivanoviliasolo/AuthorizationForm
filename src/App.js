/**
 * Является ветвителем в данном модуле, отображает 1 из 2х компонентов по условию,
 * а так же делает запрос в localStorage
 * что бы зайти->
 * файл listOfDefaultLogins.js, там можно добавить или удалить что надо.
 */

import React, { useEffect, useState } from 'react';
import { hot } from 'react-hot-loader';
import './style/App.css';
import { connect } from 'react-redux';
import LoginOrRegistration from './components/LoginOrRegistration';
import WorkedDirectory from './components/workedDirectory';

export function App({
  defLoginsState,
  dispatchSelectedLogin,
  registratedUsers,
}) {
  const [error, setError] = useState();
  const [waitRequest, setWaitRequest] = useState(false);

  /**
 * Функция, вызывающаяся только при первоначальном рендере,
 * Проверяет, имеется ли в localStorage поля с нужными ключами,
 * если имеются, то отправляет диспатч, с целью добавить эту запись(из localStorage)
 * в массив авторизированных пользователей
 *
 * @returns {dispatch || void} отправка диспатча или ничего.
 */
  useEffect(() => {
    if (localStorage.getItem('lastUserLog') !== null) {
      dispatchSelectedLogin(
        {
          type: 'ADD',
          payload: {
            login: localStorage.getItem('lastUserLog'),
            password: localStorage.getItem('lastUserPass'),
          },
        },
      );
    }
  }, []);

  /**
   * Функция которая убирает сообщение об ошибке(если таковое имеется)
   * при разлогировании пользователя
   *
   * @returns {state} - изменяет состояние error
   */
  useEffect(() => {
    setError();
  }, [registratedUsers]);

  /**
 * функция обрабатывает данные пришедшие из reduser(defLoginsState)
 * и из компонента констрактора (data) для выяснения, есть ли такой логин и пароль
 *
 * @param {array} data - данный из компонента-конструктора
 * @returns {dispatch || error(state)} возвращает диспатч если такой пароли и логин имеется
 * или выводит ошибку путем изменения состояния(фэйк задержка 1 сек).
 */
  function handleGetDataFromForm(data) {
    console.log('data',data)
    for (const state of defLoginsState) {
      let logginLegal = false;
      for (const one of data) {
        if (one.value === state.login) {
          logginLegal = true;
          continue;
        }
        if (one.value === state.password && logginLegal) {
          setWaitRequest(true);
          setTimeout(() => {
            localStorage.clear();
            localStorage.setItem('lastUserLog', state.login);
            localStorage.setItem('lastUserPass', state.password);
            dispatchSelectedLogin(
              {
                type: 'ADD',
                payload: {
                  login: state.login,
                  password: state.password,
                },
              },
            );
            setWaitRequest(false);
          }, 1000);

          return;
        }
      }
    }
    setWaitRequest(true);
    setTimeout(() => {
      setWaitRequest(false);
      setError('Имя пользователя или пароль, не правильный...');
    }, 1000);
  }

  return (
    <div>
      {Array.isArray(registratedUsers)?
      registratedUsers.length > 0
        ? <WorkedDirectory />
        : (
          <LoginOrRegistration
            header="Окно авторизации"
            closable={false}
            listOfFields={
          [
            { title: 'login', validation: 'login' },
            { title: 'password', validation: 'password' },
          ]
        }
            onSave={handleGetDataFromForm}
            error={error}
            waitRequest={waitRequest}
          />
        ):null}

    </div>
  );
}

/**
 * connect - функция обертка которая возвращает компонент переданный в возвращенную функцию
 * с пропсами, технология "react-redux"
 * @returns func
 */
export default connect(
  (state) => ({
    defLoginsState: state.login,
    registratedUsers: state.authorizedUsers,

  }),
  (dispatch) => ({
    dispatchSelectedLogin: (arg) => {
      dispatch(arg);
    },
  }),
)(

  /**
   * hot -функция обертка для пересборки проекта webpuck при сохранении файла
   * который передан в возвращаемое значение - функцию (вторые скобки)
   * @returns func
   */
  hot(module)(App),
);
