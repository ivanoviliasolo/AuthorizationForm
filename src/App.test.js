import Enzyme, {shallow, mount  } from 'enzyme';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { App } from './App';
import configureMockStore from 'redux-mock-store';

Enzyme.configure({ adapter: new Adapter() });


describe('Тестируем компонент App ',()=>{
    
    const mockStore = configureMockStore([  ]);
    const storeStateMock = {};

    const nextProps={
        defLoginsState:[],
dispatchSelectedLogin:()=>{},
registratedUsers:[ ]
}
const store = mockStore(storeStateMock);

    it('Тест компонент на наличие только 1 div ', ()=>{
        const initialProps={
            defLoginsState:[],
  dispatchSelectedLogin:()=>{},
  registratedUsers:[
            1
        ]
    }
        const AppComponent =shallow( 
                <App  {...initialProps}/>
        );
        expect(AppComponent.find('div').length).toBe(1);
    })

    it('Тестируем компонент с пустым пропсом  registratedUsers, ожидаем что выведется компонент-коснструктор регестраци\\авторизации',()=>{
       
        const AppComponent =mount ( 
                <App  {...nextProps} store={store}/>
        );
        expect(AppComponent.find('LoginOrRegistration').length).toBe(1);
    });

    it('Тестируем компонент с пустым пропсом  registratedUsers, ожидаем, что компонент'+
     'c false стороны тернарного выражения, не вызовется, соответственно мы его не найдем',()=>{

        const AppComponent =mount ( 
                <App  {...nextProps} store={store}/>
        );

        expect(AppComponent.find('WorkedDirectory').length).toBe(0);
    });

    it('Тестирование с помощью снимка, на целостность структуры компонента: ожидается совпадение снимку',()=>{
        const AppComponent =renderer.create ( 
            <App  {...nextProps} store={store}/>
    ).toJSON();

    expect(AppComponent).toMatchSnapshot();
    })

    // it('Тестируем функцию, handleGetDataFromForm, получает данные из state,( через замыкание ) '
    // +'аргумент - данные полученные через "подьем состояния" ',()=>{
    //     expect(handleGetDataFromForm('55')).toThrow('gg');
    // })
})