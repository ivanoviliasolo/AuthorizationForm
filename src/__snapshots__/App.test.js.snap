// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестируем компонент App  Тестирование с помощью снимка, на целостность структуры компонента: ожидается совпадение снимку 1`] = `
<div>
  <div
    className="main-dialog-background"
  >
    <div
      className="main-dialog-window"
    >
      <div
        className="dialog-header"
      >
        Окно авторизации
         
      </div>
      <div
        className="wrapper-near-the-all-fields"
      />
      <div
        className="main-dialog-footer"
      >
        <span
          className="users-error-footer"
        >
          
        </span>
        <button
          className="main-dialog-footer-indent"
          disabled={-1}
          onClick={[Function]}
        >
          Войти
        </button>
        <button
          onClick={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
        >
          Выйти
        </button>
      </div>
    </div>
  </div>
</div>
`;
